### ISSUE TITLE
__main__ Error: locale 'en_US' cannot be set

### ISSUE THREAD

**Issue by toxpal** [facelessuser/Rummage#246]
Recently migrated to Linux and installed Rummage today as grepwin's alternative. Installation was OK, but every time I run this app via command line, terminal has these errors:

```
notify-send 0.7.7
(__main__.py:2281): Gtk-CRITICAL **: 14:48:20.949: gtk_widget_set_allocation: assertion '_gtk_widget_get_visible (widget) || _gtk_widget_is_toplevel (widget)' failed
```

And then popup is displayed:

```
__main__ Error: locale 'en_US' cannot be set

14:42:17: Cannot set locale to language "English (U.S.)".
14:42:17: locale 'en_US' cannot be set.
```

Right after I click OK, program starts working. Is there some bug in Rummage, or I need to re-configure my system? Did some search on the Internet, and some posts suggest doing something like:

```
export LC_ALL="en_US.UTF-8"
export LC_CTYPE="en_US.UTF-8"
sudo dpkg-reconfigure locales
```

Since I'm new in Linux, not sure if I should proceed. I'm running antergos KDE and my region is set to **USA**, but other settings (like Numbers, Time, Currency...) are set to **other (non-English)** format. Maybe that's the cause of this problem?

**Comment by facelessuser** [comment]
I don't test all Linux distributions. I'm also not sure what Python version you are using. While some older release run on Python 2.7, it is recommended to use Python 3.4+ to get the latest release.


With that out of the way, the first error may simply be because an error, the utf error, occurred before there was a top window.

The second I should probably look into. I think I'm defaulting to en_US, but maybe I should just let it do whatever the systems default was. It may have been a workaround for commanding input with Unicode, but I don't remember. When I get back in town I'll take a look.

For now, if it is working after the error, if ignore it, but I'll take a look when I have a chance.

**Comment by toxpal** [comment]
Thanks for your reply. **python -V** returns **3.6.5**

Yes, it works after closing popup, just the popup itself annoys me a little :)

**Comment by facelessuser** [comment]
There are some things I might be doing that aren't necessary, maybe this is one of them. It's been a learning experience along the way. I'll definitely evaluate this and fix it if needed.

**Comment by toxpal** [comment]
After some more tests, I think it might be OS thing, not Rummage.

That's because I noticed a similar behavior with some other apps like RedNotebook and Zim. The only difference - they didn't start at all. So today I started to run them via terminal, and the same errors (locale en_US cannot be set) were displayed.

Right after I changed ALL locale/language related settings to US, Rummage (and other apps) started working perfectly. Then I did more search and found some suggestions that I might need to re-compile locale file on Linux, which I did an hour ago.

Right after that, Rummage (and other apps) starts just perfectly - no matter if system language is set to US or any other language. Did some reboots and language changes again to be sure - no problems since manual re-compilation of language file.
